classes/Store.php

```diff
@@ -136,13 +139,14 @@ public function __construct($idStore = null, $idLang = null)
      *
      * @return array|false|mysqli_result|null|PDOStatement|resource
      */
-    public static function getStores()
+    public static function getStores($idLang)
     {
         $stores = Db::getInstance()->executeS('
-            SELECT s.id_store AS `id`, s.*
+            SELECT s.id_store AS `id`, s.*, sl.*
             FROM '._DB_PREFIX_.'store s
             '.Shop::addSqlAssociation('store', 's').'
-            WHERE s.active = 1'
+            LEFT JOIN ' . _DB_PREFIX_ . 'store_lang sl ON (sl.id_store = s.id_store)
+            WHERE s.active = 1 AND sl.id_lang = ' . (int)$idLang
```

Comment:
I think this query can be optimized.

If I'm not mistaken, as is, it will bring all records from store_lang, even those that don't match the id_lang, and then process all records, filtering out those that aren't `active = 1` and from a different language. 

You should move `AND sl.id_lang = ' . (int)$idLang` into the `LEFT JOIN` `ON` clause. That way only the records from `store_lang` that match the requested language will be joined.

```
LEFT JOIN ' . _DB_PREFIX_ . 'store_lang sl ON (
    sl.id_store = s.id_store
    AND sl.id_lang = ' . (int)$idLang . '
)
WHERE s.active = 1
```